### DISCUSSION TITLE
UserLocation above ShapeSource

### DISCUSSION THREAD

**Discussion by karbone4** [rnmapbox/maps#2849] [Ideas]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### Platform

iOS, Android

### `@rnmapbox/maps` version

10.0.0-beta.78

### Standalone component to reproduce

```javascript
import React from 'react';
import {
  MapView,
  ShapeSource,
  LineLayer,
  Camera,
  UserLocation,
} from '@rnmapbox/maps';

const aLine = {
  type: 'LineString',
  coordinates: [
    [-74.00597, 40.71427],
    [-74.00697, 40.71527],
  ],
};

class BugReportExample extends React.Component {
  render() {
    return (
      <MapView style={{flex: 1}}>
        <UserLocation />
        <Camera centerCoordinate={[-74.00597, 40.71427]} zoomLevel={14} />
        <ShapeSource id="idStreetLayer" shape={aLine}>
          <LineLayer id="idStreetLayer" />
        </ShapeSource>
      </MapView>
    );
  }
}
```

### Observed behavior and steps to reproduce

`UserLocation` component is above `ShapeSource`

<img width="197" alt="image" src="https://github.com/rnmapbox/maps/assets/48047952/fc3bb3aa-268c-4e61-919a-724f94d43a69">

### Expected behavior

`UserLocation` is render before `ShapeSource` and should be behind

**Comment by mfazekas** [comment]
We don't have proper support for this at the moment. For now you can take a look at UserLocation implementation and use aboveLayerID to put stuff above our userLocation. Also not sure how to reproduce the issue with the example you posted.
